.github/workflows/docker-publish.yml

```
@@ -0,0 +1,35 @@
+name: Semantic Release
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  Release:
+    runs-on: ubuntu-latest
```

This token permission is [set by default](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) but could be useful to state it explicitly.
```suggestion
    runs-on: ubuntu-latest
    permissions:
      content: read
```